lib/apiservers/engine/backends/container.go

```diff
@@ -793,11 +793,14 @@ func (c *Container) ContainerRm(name string, config *types.ContainerRmConfig) er
 		return NotFoundError(name)
 	}
 	id := vc.ContainerID
+	secs := 0
+	running := false
 
 	// Use the force and stop the container first
-	secs := 0
 	if config.ForceRemove {
-		c.containerProxy.Stop(vc, name, &secs, true)
+		if err := c.ContainerStop(name, &secs); err != nil {
+			return err
```

Comment:
What error is returned here that _should_ prevent the deletion of the container?
I could see squashing _retries_ for Remove if we were speculatively attempting it after an error from Stop that was not related to it already being stopped, but not sure why we're not attempting it at all.

Reply:
The persona shouldn't need to know what error prevents a deletion.  An error is an error.

Reply:
Well on one hand if we get a so such container found that would be surprising and this should just return that the removal was a success in that case. If we get a InternalServiceError we might retry the stop. We could attempt everyting over again including the state check in the retry and not just the remove itself. Then the retry decider can determine whether we have see an appropriate error for return or not.

Reply:
ContainerStop makes use of the retry logic already - I'm just not sure why we're not continuing with the delete irrespective. What does ContainerStop return if the container is already stopped?

Reply:
It returns success with the id of the container.

My opinion on this is if the container could not be stopped, why are trying to do a delete?  Sounds like we're trying to take advantage of some inside knowledge of the portlayer.